src/libmongoc/tests/bsonutil/bson-parser.c

```diff
@@ -27,6 +27,7 @@ typedef enum {
    BSON_PARSER_BOOL,
    BSON_PARSER_DOC,
    BSON_PARSER_ARRAY,
+   BSON_PARSER_ARRAY_OR_DOC,
```

Comment:
This was necessary for accepting documents or pipeline arrays for update operations.